src/java.base/share/classes/java/util/UUID.java

```diff
@@ -462,31 +461,62 @@ public long node() {
      */
     @Override
     public String toString() {
-        int i0 = (int) (mostSigBits >> 32);
-        int i1 = (int) mostSigBits;
-        int i2 = (int) (leastSigBits >> 32);
-        int i3 = (int) leastSigBits;
-
         byte[] buf = new byte[36];
-        HexDigits.put4(buf, 0, i0 >> 16);
-        HexDigits.put4(buf, 4, i0);
         buf[8] = '-';
-        HexDigits.put4(buf, 9, i1 >> 16);
         buf[13] = '-';
-        HexDigits.put4(buf, 14, i1);
         buf[18] = '-';
-        HexDigits.put4(buf, 19, i2 >> 16);
         buf[23] = '-';
-        HexDigits.put4(buf, 24, i2);
-        HexDigits.put4(buf, 28, i3 >> 16);
-        HexDigits.put4(buf, 32, i3);
+
+        ByteArrayLittleEndian.setLong(buf, 0, hex8(mostSigBits >>> 32));
+        long x0 = hex8(mostSigBits);
+        ByteArrayLittleEndian.setInt(buf, 9, (int) x0);
+        ByteArrayLittleEndian.setInt(buf, 14, (int) (x0 >>> 32));
+
+        long x1 = hex8(leastSigBits >>> 32);
+        ByteArrayLittleEndian.setInt(buf, 19, (int) (x1));
+        ByteArrayLittleEndian.setInt(buf, 24, (int) (x1 >>> 32));
+        ByteArrayLittleEndian.setLong(buf, 28, hex8(leastSigBits));
+
         try {
             return jla.newStringNoRepl(buf, StandardCharsets.ISO_8859_1);
         } catch (CharacterCodingException cce) {
             throw new AssertionError(cce);
         }
     }
 
+    /**
+     * Extract the least significant 4 bytes from the input integer i, convert each byte into its corresponding 2-digit
+     * hexadecimal representation, concatenate these hexadecimal strings into one continuous string, and then interpret
+     * this string as a hexadecimal number to form and return a long value.
+     */
```

Comment:
Is there a reference describing this technique?  Such as the Hacker's Delight citation used in java.lang.Long?
If not a longer descriptive comment would be good for maintainability.

Reply:
This algorithm was researched and implemented by me, and I have added detailed description and comments